src/cpp/cuda.hpp

```diff
@@ -846,9 +850,13 @@ namespace pycuda
       virtual void detach_internal()
       {
         // Primary context comes from retainPrimaryContext.
+        // Leo: commenting out
+//#if CUDAPP_CUDA_VERSION >= 7000
         CUDAPP_CALL_GUARDED_CLEANUP(cuDevicePrimaryCtxRelease, (m_device));
+//#endif
       }
   };
+#endif
```

Comment:
It's surprising to me that you didn't also comment out `retain_primary_context` and its wrapper. How does the result even compile?